diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,183 @@
+from selenium.webdriver.support.ui import WebDriverWait
+from xblock.fields import String
+from xblockutils.base_test import SeleniumXBlockTest
+from xblockutils.resources import ResourceLoader
+import time
+# Studio adds a url_name property to each XBlock but Workbench doesn't.
+# Since we rely on it, we need to mock url_name support so it can be set via XML and
+# accessed like a normal field.
+from diagnostic_feedback.quiz import QuizBlock
+QuizBlock.url_name = String()
+
+loader = ResourceLoader(__name__)
+
+
+class DiagnosticFeedbackBaseTest(SeleniumXBlockTest):
+    """
+    The new base class for integration tests.
+    Scenarios can be loaded and edited on the fly.
+    """
+    default_css_selector = 'div.diagnostic-feedback'
+
+    def _get_previous_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[0]
+
+    def _get_previous_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#previous']")
+
+    def _get_next_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[1]
+
+    def _get_next_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#next']")
+
+    def wait_until_disabled(self, elem):
+        """ Wait until the given element is disabled """
+
+        wait = WebDriverWait(elem, self.timeout)
+        wait.until(lambda e: elem.get_attribute('class') == 'disabled',
+                   u"{} should be disabled".format(elem.text))
+
+    def wait_until_enabled(self, elem):
+        """ Wait until the given element is enabled """
+
+        wait = WebDriverWait(elem, self.timeout)
+        wait.until(lambda e: elem.get_attribute('class') == '',
+                   u"{} should be enabled".format(elem.text))
+
+    def load_scenario(self, xml_file, params=None, load_immediately=True):
+        """
+        Given the name of an XML file in the xml_templates folder, load it into the workbench.
+        """
+        params = params or {}
+        scenario = loader.render_template("xml_templates/{}".format(xml_file), params)
+        self.set_scenario_xml(scenario)
+        if load_immediately:
+            return self.go_to_view("student_view")
+
+    def load_student_view(self, xml_file, params=None, load_immediately=True):
+        """
+        Load student view in workbench
+        """
+        params = params or {}
+        scenario = loader.render_template("xml_templates/student_{}.xml".format(xml_file), params)
+        self.set_scenario_xml(scenario)
+        if load_immediately:
+            return self.go_to_view("student_view")
```

Comment:
@jareerahsan AFAICT, the only difference between this method and `load_scenario` is that it calls `loader.render_template` with slightly different arguments. Also, looking at the rest of the code, `load_student_view` is never called with custom values for `load_immediately`. You should be able to shorten it to:

```python
def load_student_view(self, xml_file, params=None):
    """
    Load student view in workbench
    """
    self.load_scenario("student_{}.xml".format(xml_file), params)
```